tests/test_sampling.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+from your_script_name import direct_sampling_with_times  # Replace with your actual script name
+import numpy as np
+
+def test_direct_sampling_with_times():
+    Q = np.array([[-1, 0.5, 0.5],
+                  [0.3, -0.8, 0.5],
+                  [0.4, 0.2, -0.6]])
+    T = 13
+    result = direct_sampling_with_times(Q, T)
+    
+    # Test for output type
+    assert isinstance(result, list), "Output should be a list"
```

Comment:
## If the result list is supposed to have a specific length, test for that
    expected_length = [the expected length]
    assert len(result) == expected_length, "The output list should have a length of expected_length"